### ISSUE TITLE
Offline install not working - always trying to connect to npmjs.org WINDOWS 7

### ISSUE THREAD

**Issue by desmond-phonegap** [arei/npmbox#34]
Same issue was raised and currently marked as closed but it has not been resolved in Windows 7 OS. https://github.com/arei/npmbox/issues/23 

As per the comments by "arie"

1). Did you untar the npmbox.npmbox file into the current directory such that ./.npmbox.cache exists? - Yes
2). Does .npmbox.cache have a ton of stuff in it including directories for npmbox? - Yes
3). Does root have permission to see inside of .npmbox.cache? - Yes
4). Did you npmbox npmbox using an older version of npmbox? (The .npmbox.cache folder named changed in version 2, older .npmbox files won't work with the newer version.) - New Version

I have tried all the steps mentioned above but still it does not install. While installing it sends a get request to https://registry.npm.js/npmjs.org/npm/-/npm-2.9.0.taz and some other 4 urls. What could be the reason it does not get from the .npmbox.cache. Thanks in advance

**Comment by willizm** [comment]
Same on me. Need help!!!

**Comment by mamboer** [comment]
Neep help too!!

**Comment by peterpuzos** [comment]
I had no issues install npmbox offline before.  But now II'm definitely having the same issue.  It keeps trying to fetch https://registry.npmjs.org/npmbox/-/npmbox-2.6.1.tgz.

Is this an issue caused by new versions of npm?

**Comment by jsmartfo** [comment]
Same issue here, upon running  

`npm install --global --cache ./.npmbox.cache --optional --cache-min 99999 --shrinkwrap false npmbox`

It looks externally, I am using node 0.12.4 and npm 2.10.1

**Comment by kienvn** [comment]
Same on me. I'm using Window 8.1 64bits

**Comment by mersed** [comment]
I am experiencing same issue on ubuntu as well. Everything is ok with permissions.

**Comment by arei** [comment]
Published 2.7.0 with a small change that I'm hoping might help this.  However, my expectations are low.  Please try it out.

If this does not resolve I am completely stumped on this and WELCOME anyone who wants to make a pull request.

I would also welcome anyone who could submit a log file... You can add `-ddd` to `npm install --global --cache ./.npmbox.cache --optional --cache-min 99999 --shrinkwrap false npmbox` before `--global` to get a lot more detail.  I recognize that the nature of this make it hard to get a log file from a remote system, but hopefully someone can help here.

**Comment by panuhorsmalahti** [comment]
I also got this when installing in Windows 2012. I'm using the latest npmbox.
```
Installing npmbox...
npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "--global" "--cache" ".\\.npmbox.cache" "--optional" "--cache-min" "99999" "--shrinkwrap" "false" "npmbox"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1
npm ERR! code ECONNRESET
npm ERR! errno ECONNRESET
npm ERR! syscall read

npm ERR! network read ECONNRESET
npm ERR! network This is most likely not a problem with npm itself
npm ERR! network and is related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network 
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
```

**Comment by dbussert** [comment]
I'm testing this by changing my registry address to a non-working server to see if npm is reaching out the internet.  `npm config set registry http://localhost`.  It appears that npm is not looking at the cache.  Also the above `npm cache add ./.npmbox.cache` fails as well with a EISDIR error.  It will add to the cache if you go into .npmbox.cache/npmbox, but even then its ignoring the cache.

```
npm info it worked if it ends with ok
npm verb cli [ '/usr/bin/node',
npm verb cli   '/usr/bin/npm',
npm verb cli   'install',
npm verb cli   '--global',
npm verb cli   '--cache',
npm verb cli   './.npmbox.cache',
npm verb cli   '--optional',
npm verb cli   '--cache-min',
npm verb cli   '99999',
npm verb cli   '--shrinkwrap',
npm verb cli   'false',
npm verb cli   'npmbox',
npm verb cli   '--loglevel',
npm verb cli   'verbose' ]
npm info using npm@2.11.3
npm info using node@v0.12.7
npm verb install initial load of /usr/lib/package.json
npm verb readDependencies loading dependencies from /usr/lib/package.json
npm verb cache add spec npmbox
npm verb addNamed "*" is a valid semver range for npmbox
npm verb addNameRange registry:http://localhost/npmbox not in flight; fetching
npm verb request uri http://localhost/npmbox
npm verb request no auth needed
npm info attempt registry request try #1 at 9:10:16 PM
npm verb request id e3a09dbcc220e80f
npm http request GET http://localhost/npmbox
npm http 404 http://localhost/npmbox
npm verb bad json <html>
npm verb bad json <head><title>404 Not Found</title></head>
```

**Comment by chrisweaver** [comment]
I had the same problem with npm 2.14.9 and node 0.12.9.

After trial and error, this worked for me:

npm install --global --cache ./.npmbox.cache --optional --cache-min 999999 --fetch-retries 0 --fetch-retry-factor 0 --fetch-retry-mintimeout 1 --fetch-retry-maxtimeout 2 --shrinkwrap false npmbox

Basically I can't stop stop npm from going to the internet, but setting the timeouts and retries to minimum causes npm to give up immediately, and then it uses the cache.

If you run the above command with -verbose you can see this bizarre behavior.

**Comment by Modarresi** [comment]
hi we are in 2016.
my npm is 3.8.3 and node 5.10.1.
i enter "npm install --global --cache .\.npmbox.cache --optional --cache-min 99999 --shrinkwrap false npmbox" but nothing hapened.
i tried to use offline-npm, but it not work on windows.
help please.

**Comment by arei** [comment]
I believe this will be fix in the forthcoming 3.0 release due to some packages having git repos instead of npm packages in them.  Closing for now, but if someone finds this again we can revisit.

**Comment by arei** [comment]
Please open a new issue and make sure to include the log any log details including the package you were trying to box/unbox, OS and version, and any other special consideration one might need to be aware of.
